uplink/src/lib.rs

```diff
@@ -105,6 +105,9 @@ pub fn stream_supergraph(
                     }
                 }
             }
+            if let Some(urls) = &urls {
+                current_url_idx = (current_url_idx + 1) % urls.len();
```

Comment:
so we re doing round robin on each url? is this our intention ? ^^ I thought we d always hit the same one, and only fallback if something fails?

Reply:
Hmm, honestly the behavior described by @o0Ignition0o if what I'd have expected.

Reply:
Hmm, honestly the behavior described by @o0Ignition0o if what I'd have expected.

Reply:
I guess that the default behavior could be seen as primitive load balancing.

Reply:
there's also a retry loop in the gateway, that will cycle between endpoints on each attempt